charts/opentelemetry-operator/templates/_helpers.tpl

```diff
@@ -142,3 +142,10 @@ Return the name of the cert-manager.io/inject-ca-from annotation for webhooks an
 {{- printf "%s/%s" .Release.Namespace (include "opentelemetry-operator.webhookCertName" .) }}
 {{- end }}
 {{- end }}
+
+{{/*
+The image to use for opentelemetry-operator.
+*/}}
+{{- define "opentelemetry-operator.image" -}}
+{{- printf "%s:%s" .Values.manager.image.repository (default .Chart.AppVersion .Values.manager.image.tag) }}
```

Comment:
I am missing what this template is doing differently that the existing logic

Reply:
there is no difference

Reply:
But how does this change help the issue described in the PR? Also, our chart explicitly does not support `1.23`, so I don't really want to add features for that version.

Reply:
I'm sorry, my description above is a little rough.

@TylerHelmuth  I also agree that we shoul not do multi-version(1.23) compatibility in the community upstream. But, the downstream can impl multi-version compatibility in the following ways:

1. use `otel-operator` as sub-chart:

```bash
dependencies:
  - name: opentelemetry-operator
    version: 0.58.0
    repository: https://open-telemetry.github.io/opentelemetry-helm-charts
```

2. then they can override `{{- define "opentelemetry-operator.image" -}}` logic in there own parent template(`_helpers.tpl`) to override the value of the operator image:

```bash
{{- define "opentelemetry-operator.image" -}}
{{- $suggest_tag := .Chart.AppVersion }}
  {{- if (semverCompare "< 1.23-0" .Capabilities.KubeVersion.Version) }}
    {{- $suggest_tag = 0.85.0 }}
  {{- else }}
    {{- $suggest_tag = 0.99.0 }}
  {{- end }}

{{- printf "%s:%s" .Values.opentelemetry-operator.manager.image.repository $suggest_tag }}
{{- end }}
```

Reply:
>  override {{- define "opentelemetry-operator.image" -}}
 woah, I actually didn't know this was possible.